blaze/expr/collections.py

```diff
@@ -31,19 +33,200 @@
            'isin',
            'Join',
            'join',
+           'Key',
            'Merge',
            'merge',
            'Sample',
            'sample',
            'Shift',
            'shift',
            'Sort',
+           'Sort2',
            'sort',
+           'sort2',
            'Tail',
            'tail',
            'transform']
 
 
+class Key(Expr):
+
+    """Expression sorted in ascending or descending order"""
+
+    _arguments = '_child', 'table', '_key', 'ascending'
+
+    def _dshape(self):
+        return self._child.dshape
+
+    @property
+    def key(self):
+        return list(self._key)
+
+    def _len(self):
+        return self._child._len()
+
+    @property
+    def _name(self):
+        return self._child._name
+
+    def _copy(self):
+        """Create a copy of self, detached from any specific table."""
+        return Key(None, None, self._key, self.ascending)
+
+    def _invert(self):
```

Comment:
it is only reserved for blaze exprs to dynamically update the `__invert__` method.